### PR TITLE
[Keyboard] Add Royal Kludge RKG-68

### PR DESCRIPTION
**Pull request by Thomaz-Peres**
## Description

Create keyboard map for Royal Kludge RKG-G68

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rkg68/info.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    "url": "https://bit.ly/3JI7KUg",
+    "maintainer": "Thomaz-Peres",
+    "usb": {
+        "vid": "0x05C4",
+        "pid": "0x05C4",
+        "device_version": "1.0.0"
+    },
+	"processor": "atmega32u4",
+    "bootloader": "atmel-dfu",
+    "layouts": {
```
Comment (zvecr) [inline keyboards/rkg68/info.json]:
```suggestion
    "matrix_pins": {
        "cols": ["F0", "F1", "E6", "C7", "C6", "B6", "D4", "B1", "B7", "B5", "B4", "D7", "D6", "B3", "F4"],
        "rows": ["D0", "D1", "D2", "D3", "D5"]
    },
    "diode_direction": "COL2ROW",
    "layouts": {
```

File: keyboards/rkg68/info.json [hunk 2]
```diff
@@ -0,0 +1,115 @@
+{
+	"manufacturer": "Royal Kludge",
+	"keyboard_name": "rkg68",
+	"maintainer": "Thomaz-Peres",
+	"bootloader": "atmel-dfu",
+	"config_h_features": {
+		"encoder": false
+	},
```
Comment (zvecr) [inline keyboards/rkg68/info.json:8]: Copying in `qmk info` content is wrong here, as api specific content should not exist in the repo

File: keyboards/rkg68/info.json [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	"features": {
+		"encoder": false
+	},
+	"keyboard_folder": "rkg68",
+	"matrix_pins": {
+		"cols": [ "F0", "F1", "E6", "C7", "C6", "B6", "D4", "B1", "B7", "B5", "B4", "D7", "D6", "B3", "F4" ],
+		"rows": [ "D0", "D1", "D2", "D3", "D5" ]
+		},
+		"platform_key": "avr",
```
Comment (zvecr) [inline keyboards/rkg68/info.json:21]:
```suggestion
```

File: keyboards/rkg68/info.json [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+		"processor": "atmega32u4",
+		"processor_type": "avr",
```
Comment (zvecr) [inline keyboards/rkg68/info.json:23]:
```suggestion
```

File: keyboards/rkg68/info.json [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		"encoder": false
+	},
+	"diode_direction": "COL2ROW",
+	"encoder": {
+		"enabled": false
+	},
+	"features": {
+		"encoder": false
+	},
+	"keyboard_folder": "rkg68",
```
Comment (zvecr) [inline keyboards/rkg68/info.json:16]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/rkg68/readme.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# RK G68
+
+![Keyboard picture](https://imgur.com/UnOJPZr.jpg)
```
Comment (drashna) [inline keyboards/rkg68/readme.md:3]:
```suggestion
![Keyboard picture](https://imgur.com/UnOJPZrh.jpg)
```

File: keyboards/rkg68/readme.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+- **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+- **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
+- **Keycode in layout**: Press the key `FN` and `Left CTRL`.
+
+## Layouts
+
+- ANSI layout;
+
+![image](https://user-images.githubusercontent.com/58439854/225489922-a6a4df65-aff7-4835-a452-3b1afdb6fc15.png)
```
Comment (drashna) [inline keyboards/rkg68/readme.md:38]: Please rehost this image on Imgur.

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/rkg68/readme.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    make rkg68:default
+
+Flashing example for this keyboard:
+
+    make rkg68:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+---
```
Comment (zvecr) [inline keyboards/rkg68/readme.md:26]:
```suggestion
```

File: keyboards/rkg68/readme.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+* Keyboard Maintainer: [Royal Kludge](https://rkgamingstore.com/)
+* Hardware Supported: Royal Kludge RK G68
+* Hardware Availability: [RKG 68](http://en.rkgaming.com/download/1/)
+
+## Availability
+
+The RKG68 is available for purchase through [AliExpress](https://www.aliexpress.com/).
+
+## How to compile
+
```
Comment (zvecr) [inline keyboards/rkg68/readme.md:14]:
```suggestion
```

File: keyboards/rkg68/info.json [hunk 10]
```diff
@@ -0,0 +1,109 @@
+{
+	"manufacturer": "Royal Kludge",
+	"keyboard_name": "rkg68",
+	"maintainer": "Thomaz-Peres",
+	"bootloader": "atmel-dfu",
+	"processor": "atmega32u4",
+	"diode_direction": "COL2ROW",
+	"encoder": {
+		"enabled": false
+	},
```
Comment (zvecr) [inline keyboards/rkg68/info.json:10]:
```suggestion
```

File: keyboards/rkg68/info.json [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+	"features": {
+		"encoder": false
```
Comment (zvecr) [inline keyboards/rkg68/info.json:12]:
```suggestion
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true
```

File: keyboards/rkg68/info.json [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+	"rgblight": {
+		"animations": {
+			"knight": true,
+			"rainbow_swirl": true
+		},
+		"brightness_steps": 17,
+		"hue_steps": 10,
+		"led_count": 4,
+		"pin": "B3",
+		"saturation_steps": 17
```
Comment (zvecr) [inline keyboards/rkg68/info.json:27]:
```suggestion
		"led_count": 4,
		"pin": "B3"
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/rkg68/info.json [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+	"url": "http://en.rkgaming.com/",
+	"usb": {
+		"device_version": "1.0.0",
+		"pid": "0x0049",
+		"vid": "0x0049"
+	},
+	"layouts": {
+		"LAYOUT": {
+			"layout": [
+				{"matrix": [0, 0], "x": 0, "y": 0.0},
```
Comment (fauxpark) [inline keyboards/rkg68/info.json:34]:
```suggestion
				{"matrix": [0, 0], "x": 0, "y": 0},
```
etc

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Comment by darknblack** [comment]
Hi @Thomaz-Peres, 

I have RKG68 and my keyboard layout matches exactly from the one on the readme.md, however my keyboard has different PID (00E7). Do you think flashing it by simply changing the PID from the `info.json` will do? If not maybe you could guide me or at least provide images of the board you have. I will really appreciate your help. 

Thank you so much

**Comment by Thomaz-Peres** [comment]
Hello @darknblack.

In response to your question, if you update the info.json file with the PID, it will probably work (make sure to check the bootloader, processor, and the VID as well).
If it doesn’t work, and you want, we can discuss this on QMK Discord. 
If your solution works, you can submit a PR to update the PID, and if there’s more new information, you can update that too.

##### Why PID is different ?
This is because (If I’m not mistaken), RoyalKludge updated the RKG68 with new PCBs. (I really struggled to find information about the old one).

**NOTE**: It’s been a few months since I last read about keyboards, so some of the information I provide might be incorrect (apologies in advance)

**Comment by darknblack** [comment]
Hi @Thomaz-Peres,

> (make sure to check the bootloader, processor, and the VID as well)
Is there a way to know the bootloader without opening the keyboard?

Here's the device info from Device Manager:
`Device HID\VID_258A&PID_00E7&MI_00\b&110c8674&0&0000 was configured.`

QMK Toolbox:
`USB device connected (HidUsb): (Standard system devices) USB Input Device (258A:00E7:1007)`

Thank you so much for your help 🙇 🤞

**Comment by HorrorTroll** [comment]
> Here's the device info from Device Manager: `Device HID\VID_258A&PID_00E7&MI_00\b&110c8674&0&0000 was configured.`
> 
> QMK Toolbox: `USB device connected (HidUsb): (Standard system devices) USB Input Device (258A:00E7:1007)`

If the keyboard support bluetooth, it mean the MCU is not supported for QMK. Since official QMK was never supported wireless.

**Comment by darknblack** [comment]
Hi @HorrorTroll

> If the keyboard support bluetooth, it mean the MCU is not supported for QMK. Since official QMK was never supported wireless.

Does that mean QMK will not work in general for keyboards with Bluetooth feature? Or you can still flash the firmware but it will only work on wired mode? Thank you

**Comment by Thomaz-Peres** [comment]
@darknblack, I'm not familiar with these new RK versions, but typically, you can accomplish this using a macro (in the oldest version, I used fn+shift+b or fn+shift+enter for 3 seconds, although I don't remember exactly). Therefore, you can update the VID and the PID with the Device Manager you've provided.

@HorrorTroll, but does using the USB work, right? I don't recall reading anything about it; I understood that only when using the keyboard as Bluetooth it would not work. If you have a link to read about this, please share.

**Comment by HorrorTroll** [comment]
> Does that mean QMK will not work in general for keyboards with Bluetooth feature? Or you can still flash the firmware but it will only work on wired mode? Thank you
> But does using the USB work, right? I don't recall reading anything about it; I understood that only when using the keyboard as Bluetooth it would not work. If you have a link to read about this, please share.

Sorry for late reply to you guys. As I recall that RK and other company are order the OEM PCB directly from Evision/HFD or other manufacturer for wireless support. Those company are using like Sonix SN32F248B, CH555 or SH68F90 at the main MCU. Even trying to port QMK and use only wired is not really possible.

At for the Sonix SN32F248B, it kinda really on old PCB. So I don't know if RK are even have those anymore. But luckily that these chip are having community port QMK on it. It called SonixQMK.

While CH555 and SH68F90 are Intel 8051 (8-bit Enhanced customize) architecture, so yeah. It not possible to do that anyway.

**Comment by Thomaz-Peres** [comment]
That's nice, I found the repository here! I will do some tests to see if it works.

And thank you very much, I really didn't know about that; probably because of this, I've had many problems!

**Comment by darknblack** [comment]
Hello, I recently opened up my keyboard to install new switches, and I discovered that it uses the following chip:

`BYK916-00997A NRW37-1 2203-D`

I'm not sure if this chip is currently supported. Could someone please confirm this? Thank you

**Comment by fauxpark** [comment]
BYK916 is 8051 architecture. You will not be able to run QMK on this.

**Comment by karror** [comment]
Hello @Thomaz-Peres! Thank you for your job for community! I know that was a while when these thread was alive, but i want to know if You had some success on flashing rkg68 with this firmware. I ask, because I found it on production in qmk compiler list of keyboards. My issue is that, I bought "rk61 plus" which have this "amazing" chip byk916, and i find out that rkg68 is based on the same family from sinowealth. Thats [here](https://github.com/carlossless/sinowealth-kb-tool?tab=readme-ov-file#keyboards). It's also have bluetooth and 2.4G interfaces. I wondering that, if your firmware is working, could that be, your driver will handle also byk916 (SH68F90A) of "rk61 plus"?

**Comment by fauxpark** [comment]
QMK will not run on this board or MCU. Please read what I have posted above.

**Pull request merged**